enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,305 @@
+
+If `platform.gcp.userLabels` of install-config has any label defined, same will be added 
+to all the gcp resources created by Openshift except, when the label validation fails 
+to adhere to any of the below conditions
+1. A label name must have minimum of 1 character and can have maximum of 63 characters.
+2. A label value can have maximum of 63 characters.
+3. A label name and value must contain only lowercase letters, numeric characters, 
+   underscores, and dashes.
+4. A label name must start with a lowercase letter.
+5. Each resource can have multiple labels, up to a maximum of 50.
+   - GCP supports a maximum of 64 labels per resource. Restricting the number of 
+    user defined labels to 50 and reserving 14 for Openshift's internal use.
+
+All in-cluster operators that create GCP resources (Cluster Infrastructure ,Storage ,Node
+,NetworkEdge, Internal Registry ,CCO) will apply these labels during resource creation.
+
+The userLabels field is intended to be set at install time and is considered immutable. 
+Components that respect this field must only ever add labels that they retrieve from this 
+field to cloud resources, they must never remove labels from the existing underlying cloud 
+resource even if the labels are removed from this field(despite it being immutable).
+
+If the userLabels field is changed post-install, there is no guarantee about how an 
+in-cluster operator will respond to the change. Some operators may reconcile the 
+change and update the labels on the GCP resource. Some operators may ignore the change. 
+However, if labels are removed from userLabels, the label will not be removed from the 
+GCP resource.
+
+### Workflow Description
+
+- An Openshift administrator requests to add required labels to all GCP resources 
+  created by Openshift by adding it in `platform.gcp.userLabels`
+- Openshift installer validates the labels defined in `platform.gcp.userLabels` and 
+  adds these labels to all resources created during installation and also updates 
+  `.status.platformStatus.gcp.resourceLabels` of the `infrastructure.config.openshift.io`
+- In cluster operators refers `.status.platformStatus.gcp.resourceLabels` of the 
+  `infrastructure.config.openshift.io` to add labels to the resources created later.
+
+#### Variation [optional]
+
+### API Extensions
+Enhancement requires below modifications to the mentioned CRDs
+- Add `userLabels` field to `platform.gcp` of the `installconfigs.install.openshift.io`
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: installconfigs.install.openshift.io
+spec:
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        properties:
+          platform:
+            properties:
+              gcp:
+                properties:
+                  userLabels:
+                    additionalProperties:
+                      type: string
```

Comment:
same comment as azure: not sure about the type here. it should be the equivalent of a map in go

Reply:
Yes, it will be of type `map[string]string`